oauth2/oauth2-resource-server/src/test/java/org/springframework/security/oauth2/core/expression/OAuth2AuthenticationSupport.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2002-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.security.oauth2.core.expression;
+
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.oauth2.resourceserver.authentication.AbstractOAuth2AccessTokenAuthenticationToken;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * @author Josh Cummings
+ */
+public class OAuth2AuthenticationSupport {
```

Comment:
I feel that this class is not needed. I don't think it's much work to `setup` an `AbstractOAuth2AccessTokenAuthenticationToken` in the test itself.

Reply:
See https://github.com/jzheaux/spring-security-oauth2-resource-server/issues/23

Reply:
I guess that is a difference of opinion. I'm inclined to keep and then replace with #23, as Rob alluded to.

Reply:
Makes sense to replace it with #23